### PR TITLE
fix: playToggle state incorrect when autoplay is blocked in Firefox

### PR DESCRIPTION
**Pull request by alex-barstow**
See code comment for explanation.

### CODE REVIEW

File: src/playMiddleware.js [hunk 1]
```diff
@@ -55,13 +55,12 @@ obj.playMiddleware = function(player) {
...
-      // We fingerprint Chrome using e.message and send a pause for consistency.
-      // This keeps the play button synchronized if play is rejected.
+      // Safari issues a pause event when autoplay is blocked but Chrome and Firefox
+      // do not, so we send a pause for consistency in those cases. This keeps the
+      // play button in the correct state if play is rejected.
       } else if (playPromise && playPromise.catch) {
         playPromise.catch((e) => {
-          if (e.message === 'play() failed because the user didn\'t interact with the ' +
-            'document first. https://goo.gl/xX8pDD') {
+          if (e.name === 'NotAllowedError' && (videojs.browser.IS_CHROME || videojs.browser.IS_FIREFOX)) {
```
[the remark this answers is not in the document]
Comment (gkatsev) [reply src/playMiddleware.js]: I am not sure we can avoid user-agent detection/matching the error specifically. We're working around specific browser behavior. We should try and preserve the browser behavior here if we can, unless we have a separate say of fixing it.

[the remark this answers is not in the document]
Comment (alex-barstow) [reply src/playMiddleware.js]: I'm looking into the feasibility of removing [this play trigger](https://github.com/videojs/videojs-contrib-ads/blob/master/src/playMiddleware.js#L52). Currently, this behavior (autoplay rejected -> trigger `play` anyways) does not match the video element behavior when autoplay is rejected (no play event) so we could possibly justify removing it for this reason, however removing it breaks a lot so more extensive refactoring would be necessary.

[the remark this answers is not in the document]
Comment (gkatsev) [reply src/playMiddleware.js]: I think that's what Safari does.

[the remark this answers is not in the document]
Comment (alex-barstow) [reply src/playMiddleware.js]: Safari fires a `pause` event when autoplay is rejected but not a `play` event.

**Comment by alex-barstow** [comment]
Tests are fixed. And following offline discussion, we decided a user agent check may be practically unavoidable in this case.

**Pull request merged**